src/charm.py

```diff
@@ -100,6 +100,8 @@ def __init__(self, *args):
             redis_relation={},
         )
         self._require_nginx_route()
+        self._gems_installed = False
+        self.framework.observe(self.on.install, self._on_install)
```

Comment:
One thing to note here is that this event won't be called if the workload container is restarted without the charm container being restarted. The only hook that's executed in that case is the discourse_pebble_ready hook. This would likely work okay because you're calling the `_on_install` hook as part of `_config_changed` (which is called by the discourse_pebble_ready event handler), but it makes it feel like we should have a layer of abstraction for the thing we're doing (installing gems) and call that from the relevant event rather than calling `self._on_install(event)` (but seem the comment above about where this should be called from).

Reply:
My idea was to add only the install event but then I noticed that pebble ready event sometimes is fired first 🤔
```Note that although juju guarantees to fire install before pebble-ready, it is possible that the container is already ready by the time install fires, and can_connect would return True.```
https://juju.is/docs/sdk/container-name-pebble-ready-event

So that's why I added it to the config_changed method. But ideally, this shouldn't be needed...

Anyway, I added this variable just to see if it works, I believe it would be better/safer to check for the directory content.